frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -999,12 +1000,14 @@ final class BloopBspServices(
         projects: Seq[ProjectMapping],
         state: State
     ): BspResult[bsp.ScalacOptionsResult] = {
+      val uris = mutable.Map.empty[Path, bsp.Uri]
```

Comment:
Are we guaranteed that calls to the `scalacOptions` can not happen concurrently? If not, we should use a `TrieMap` instead of a `mutable.Map`.

Reply:
Yes, were guaranteed this is the case here, were only caching inside the method that handles one request